### ISSUE TITLE
[BUG] Destroying a release does not destroy the corresponding cache directory (ZFS)

### ISSUE THREAD

**Issue by joh-ku** [BastilleBSD/bastille#715]
**[MANDATORY] Describe the bug [MANDATORY]**
When running `bastille destroy RELEASE` (ZFS) the corresponding cache directory remains untouched.

**[MANDATORY] Bastille and FreeBSD version (paste ``bastille -v && freebsd-version -kru`` output)**
0.10.20231125
14.1-RELEASE-p3
14.1-RELEASE-p3
14.1-RELEASE-p3

**[MANDATORY] How did you install bastille? (port/pkg/git)**
pkg

**[optional] Steps to reproduce?**
1. Run e.g. `bastille destroy 14.0-RELEASE` (or any release version you may want to destroy)
2. /usr/local/bastille/cache/14.0-RELEASE still exists

**[optional] Expected behavior**
When a release is destroyed, its corresponding cache directory should be deleted as well.

**Comment by foudfou** [comment]
This is not well documented but the cache is deleted on `force`: `bastille destroy force 14.0-RELEASE` .

**Comment by joh-ku** [comment]
@foudfou correct, there is a force flag. Thanks for pointing this out!

The question for me is, if an additional flag is required at all and hence, if "destroy" should delete the release and cache directories all together. Pulling the image again is cheap anyway.

**Comment by foudfou** [comment]
Pulling the image is not so cheap to be honest (minutes) compared to creating a release from the cache (seconds). Plus upstream might not have the locally destroyed release anymore (which will probably make jails hard to use).

That said, I would also expect destroying a release to also clean up the cache and will amend the PR accordingly.

**Comment by joh-ku** [comment]
Well, not as cheap as serving it from the cache, agreed. Thank you for the PR! I think the behavior is clearer now.

**Comment by bmac2** [comment]
@cedwards  what do you think the desired action is here??  Isn't the way itworks not destroying cache the designed way, or should it be changed?  Don't want to make changes without your input.

**Comment by marschro** [comment]
Another thing to keep in mind - ist it true, that the zfs filesystem for that release then still does exist? 
If so, then maybe it can be considered to "clean-up" that too?

**Comment by michael-o** [comment]
Has this one been resolved with the last release?

**Comment by bmac2** [comment]
I do not believe this one has been addressed yet.  No PR in this release addressed this.

@michael-o

**Comment by michael-o** [comment]
> I do not believe this one has been addressed yet. No PR in this release addressed this.
> 
> [@michael-o](https://github.com/michael-o)

Alright,  am still questioning why force would be useful unless something is mounted and needs to be taken down. Cache should always go with.

**Comment by tschettervictor** [comment]
This is documented, but should we just delete everything including the cache? 

Or are we good with documentation on it?

**Comment by joh-ku** [comment]
I don't think the force option provides significant benefit in this context and in this sense I'd be fine with deleting release and cache together. Anyway, I'm definitely fine with what the majority thinks is best.

**Comment by tschettervictor** [comment]
@michael-o @bmac2

**Comment by bmac2** [comment]
I think documenting the behavior and using -f if you want to destroy cache is the way to go.  Not everyone wants to destroy the cache and the image is not always cheap to download.

Anyone disagree at this point?

**Comment by michael-o** [comment]
> I think documenting the behavior and using -f if you want to destroy cache is the way to go. Not everyone wants to destroy the cache and the image is not always cheap to download.
> 
> Anyone disagree at this point?

I personally think that force applies to unmount and rm only. Everything what can be deleted should be deleted. One could add a flag to retain the cache.

**Comment by tschettervictor** [comment]
Perhaps a --no-cache flag?

**Comment by bmac2** [comment]
ok so on a destroy WITHOUT any flags we set it to delete the cache.  Anyone against this method?
